### PR TITLE
feat: add vue-recommended/element-attribute-order rule w tests

### PR DESCRIPTION
**Pull request by David-Pena**

### CODE REVIEW

File: src/rules/vue-recommended/elementAttributeOrder.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+const reportElementAttributeOrder = () => {
+  if (elementAttributeOrderFiles.length > 0) {
+    // count only non duplicated objects (by its `filename` property)
+    const fileCount = getUniqueFilenameCount<ElementAttributeOrder>(elementAttributeOrderFiles, 'filename')
+
+    console.log(`\n${TEXT_INFO}vue-recommended${TEXT_RESET} ${BG_ERR}element attribute order ${BG_RESET} detected in ${fileCount} files.`)
+    console.log(
+            `👉 ${TEXT_WARN}Whenever a build system is available to concatenate files, each component should be in its own file.${TEXT_RESET} See: https://vuejs.org/style-guide/rules-strongly-recommended.html#component-files`,
```
Comment (rrd108) [inline src/rules/vue-recommended/elementAttributeOrder.ts:74]: https://vuejs.org/style-guide/rules-recommended.html#element-attribute-order isn't the right link here?

File: src/rules/vue-recommended/elementAttributeOrder.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  'v-html',
+  'v-text',
+]
+
+const checkElementAttributeOrder = (template: SFCTemplateBlock | null, filePath: string) => {
+  if (!template)
+    return
+
+  // Remove the <template> tags to avoid checking them
+  const innerTemplate = template.content.replace(/<\/?template>/g, '')
```
Comment (rrd108) [inline src/rules/vue-recommended/elementAttributeOrder.ts:34]: Can you have magic-regexp here for any of the regexes?

**Review by rrd108 — CHANGES_REQUESTED** [review]

File: src/rules/vue-recommended/elementAttributeOrder.ts (hunk 1 above)
Comment (David-Pena) [reply src/rules/vue-recommended/elementAttributeOrder.ts]: fixed in the last commit

File: src/rules/vue-recommended/elementAttributeOrder.ts (hunk 2 above)
Comment (David-Pena) [reply src/rules/vue-recommended/elementAttributeOrder.ts:34]: I tried, but the result from `convert` was not working as expected. So I think in a next iteration of the rule, I'd have more knowledge with `magic-regexp`

**Review by rrd108 — CHANGES_REQUESTED** [review]
The text of the cli output is also to be changed, right?

**Comment by David-Pena** [comment]
> The text of the cli output is also to be changed, right?

Fixed in the last commit 😅 my bad

**Review by rrd108 — APPROVED** [review]

**Pull request merged**